### DISCUSSION TITLE
Move Cursor to Specified Position for Keyboard-Controlled Mouse

### DISCUSSION THREAD

**Discussion by WantenMN** [YaLTeR/niri#778] [Q&A]
Does Niri have a built-in function to move the cursor to a specified position, like Hyprland's `hyprctl dispatch movecursor x y`? I'm looking for a solution to control the mouse via keyboard with a custom script. I've tried xdotool, xautomation, ydotool, and warpd, but they either don't work or aren't intuitive.

**Comment by YaLTeR** [comment]
It doesn't, though #630 provides wlr-virtual-pointer which might be useful for this (idk if there are any clients that make use of it). Also I guess it shouldn't be too difficult to implement via IPC, but this invites a whole host of other IPC like pressing mouse buttons, pressing keyboard buttons, and so on, for which the virtual input device protocols already exist.
